docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md

```diff
@@ -0,0 +1,211 @@
++               url: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               guid: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               custom_elements: [{ "content:encoded": edge.node.body }],
++             })
++           ),
++         query: `
++         {
++           allMdx(
++             limit: 1000,
++             sort: { order: DESC, fields: [frontmatter___date] },
++           ) {
++             edges {
++               node {
++                 fields { slug }
++                 frontmatter {
++                   title
++                   date
++                 }
++                 body
++               }
++             }
++           }
++         }
++         `,
++         output: `/rss.xml`,
++         title: `Gatsby RSS feed`,
++       },
++     ],
++   },
++ },
+```
+
+### Step 3
+
+In `gatsby-node.js`, change `allMarkdownRemark` to `allMdx` and `MarkdownRemark` to `Mdx`.
+
+```diff:title=gatsby-node.js
+# line 11
+- allMarkdownRemark(
++ allMdx(
+
+# line 35
+- const posts = result.data.allMarkdownRemark.edges
++ const posts = result.data.allMdx.edges
+
+# line 56
+- if (node.internal.type === `MarkdownRemark`) {
++ if (node.internal.type === `Mdx`) {
+```
+
+### Step 4
+
+In `src/pages/index.js`, replace `allMarkdownRemark` with `allMdx` in both the `render()` method and also the GraphQL query.
+
+```diff:title=src/pages/index.js
+# line 13
+- const posts = data.allMarkdownRemark.edges
++ const posts = data.allMdx.edges
+
+# line 59
+- allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
++ allMdx(sort: { fields: [frontmatter___date], order: DESC }) {
+```
+
+### Step 5
+
+In `src/templates/blog-post.js`, add an import statement for `MDXRenderer` from `gatsby-plugin-mdx` and replace `markdownRemark` with `mdx` in both the `render()` method and also the GraphQL query. Also, we'll be using `body` instead of `html` in our GraphQL query.
+
+```diff:title=src/templates/blog-post.js
+# line 3
++ import { MDXRenderer } from "gatsby-plugin-mdx"
+
+# line 11
+- const post = this.props.data.markdownRemark
++ const post = this.props.data.mdx
+
+# line 41
+- <section dangerouslySetInnerHTML={{ __html: post.html }} />
++ <MDXRenderer>{post.body}</MDXRenderer>
+
+# line 93
+- markdownRemark(fields: { slug: { eq: $slug } }) {
++ mdx(fields: { slug: { eq: $slug } }) {
+
+# line 96
+- html
++ body
+```
+
+And... that's it! After these changes, a Gatsby blog should be able to use MDX files to render JSX with markdown! To test, add a `.mdx` file to `content/blog/` and check if the JSX renders.
+
+For example, the following code should render a test button.
+
+````mdx:title=content/blog/example.mdx
+---
+title: MDX!
+date: "2019-10-22"
+description: "A post showing MDX in action"
+---
+
+This is a post showing MDX in action. This starter now comes with MDX out-of-the-box!
+
+```js
+// you can write JSX in your Markdown!
+<button>test</button>
+```
+
+<button>test</button>
+
```

Comment:
Maybe tell them how to see this example? Run `gatsby develop` and navigate where?

Reply:
done